rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMNodeData.java

```diff
@@ -232,6 +261,72 @@ public void setJvmName(String jvmName) {
         this.jvmName = jvmName;
     }
 
+    public boolean getLocked() {
+        return locked;
+    }
+
+    public void setLocked(boolean locked) {
+        this.locked = locked;
+    }
+
+    @Column(length = Integer.MAX_VALUE)
+    @Type(type = "org.hibernate.type.SerializableToBlobType", parameters = @Parameter(name = SerializableToBlobType.CLASS_NAME, value = "java.lang.Object"))
+    public Client getLockedBy() {
+        return lockedBy;
+    }
+
+    public void setLockedBy(Client lockedBy) {
+        this.lockedBy = lockedBy;
+    }
+
+    public long getLockTime() {
+        return lockTime;
+    }
+
+    public void setLockTime(long lockTime) {
+        this.lockTime = lockTime;
+    }
+
+    @Column(length = Integer.MAX_VALUE)
+    @Type(type = "text")
+    public String getCommandLine() {
+        return commandLine;
+    }
+
+    public void setCommandLine(String commandLine) {
+        this.commandLine = commandLine;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setdescription(String description) {
+        this.description = description;
+    }
+
+    public RMNodeEvent createNodeEvent(String nodeSourceName) {
+
+        RMNodeDescriptor rmNodeDescriptor = new RMNodeDescriptor();
+        rmNodeDescriptor.setNodeURL(nodeUrl);
+        rmNodeDescriptor.setDefaultJMXUrl(jmxUrls == null ? null : jmxUrls[JMXTransportProtocol.RMI.ordinal()]);
+        rmNodeDescriptor.setDescriptorVMName(jvmName);
+        rmNodeDescriptor.setHostName(hostname);
+        rmNodeDescriptor.setLocked(locked);
+        rmNodeDescriptor.setLockedBy(getLockedBy() == null ? null : getLockedBy().getName());
+        rmNodeDescriptor.setLockTime(getLockTime());
+        rmNodeDescriptor.setNodeSourceName(nodeSourceName);
+        rmNodeDescriptor.setOwnerName(getOwner() == null ? null : getOwner().getName());
+        rmNodeDescriptor.setProactiveJMXUrl(jmxUrls == null ? null : jmxUrls[JMXTransportProtocol.RO.ordinal()]);
+        rmNodeDescriptor.setProviderName(getProvider() == null ? null : getProvider().getName());
+        rmNodeDescriptor.setState(getState());
+        rmNodeDescriptor.setStateChangeTime(getStateChangeTime());
+        rmNodeDescriptor.setVNodeName(jvmName);
+
+        RMNodeEvent rmNodeEvent = new RMNodeEvent(rmNodeDescriptor, null, null, null);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "rmNodeEvent". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)